Content.Server/Paper/PaperSystem.cs

```diff
@@ -124,6 +129,11 @@ private void OnInputTextMessage(EntityUid uid, PaperComponent paperComp, PaperIn
             UpdateUserInterface(uid, paperComp);
         }
 
+        private void OnWrittenWith(EntityUid uid, ActivateOnPaperOpenedComponent comp, ref PaperWriteEvent args)
+        {
+            RaiseLocalEvent(uid, new UseInHandEvent(args.User));
```

Comment:
use `SharedInteractionSystem.TryUseItemInHand()` or .UseInHandInteraction() 
Remember to rename the event too  (also, now that i expanded, just above this is a PaperInputTextMessage, cant you subscribe to that instead?)

Reply:
ill test when inputtext fires

Reply:
also uh this would let you press X then write something to validcheck i think if i did both changes

Reply:
> also uh this would let you press X then write something to validcheck i think if i did both changes

how would you valid check it without triggering it in your hands?

Reply:
1. hold pen and bike horn
2. open paper
3. press x
4. start writing something
5. honk means pen is bomb, tatoralert.ogg

Reply:
Ah, i see, well this is a bug on master since you can also open the UI, drop the pen and continue to write. The fix should be making a check in the DidUnequipHandEvent and update the paper UI accordingly (both displaying a message that a pen is needed in one of the hands both in the UI and a popup) and then loop through all hands to find a pen in the ActivateOnPaperOpenedComponent handler (could make this last step not necessary if we mandate that the pen MUST be inhands to write and then subscribe and make checks in HandDeselectedEvent, tho im not so sure about this)

I wont require you to do this tho and valid checking pens isnt that big of a deal, just activate in PaperInputTextMessage for now if you want and the issue will be fixed along the others (could be partially fixed by looping through hands and checking if the pen is there)

Reply:
PaperInputTextMessage has no clean way to get the pen used to write with, will stick to current thing but using better activation